govcd/api_token.go

```diff
@@ -155,7 +241,163 @@ func (vcdClient *VCDClient) GetBearerTokenFromApiToken(org, token string) (*type
 		// If decoding the error fails, we return the raw body (possibly an unencoded internal server error)
 		return nil, fmt.Errorf("access token retrieved from API token was empty - %s %s", resp.Status, string(body))
 	}
-	return &tokenDef, nil
+
+	return newToken, nil
+}
+
+func (client *Client) doTokenRequest(org, endpoint, apiVersion, contentType string, data io.Reader) (*http.Response, error) {
+	newUrl := url.URL{
+		Scheme: client.VCDHREF.Scheme,
+		Host:   client.VCDHREF.Host,
+	}
+
+	userDef := "tenant/" + org
+	if strings.EqualFold(org, "system") {
+		userDef = "provider"
+	}
+
+	reqHref, err := url.ParseRequestURI(fmt.Sprintf("%s/oauth/%s/%s", newUrl.String(), userDef, endpoint))
+	if err != nil {
+		return nil, fmt.Errorf("error getting request URL from %s : %s", reqHref.String(), err)
+	}
+
+	req := client.NewRequest(nil, http.MethodPost, *reqHref, data)
+	req.Header.Add("Accept", fmt.Sprintf("application/*;version=%s", apiVersion))
+	req.Header.Add("Content-Type", contentType)
+
+	resp, err := client.Http.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("error getting request URL from %s : %s", reqHref.String(), err)
+	}
+
+	return checkRespWithErrType(types.BodyTypeJSON, resp, err, &types.OpenApiError{})
+}
```

Comment:
Can this also be done by some generic function in `openapi.go`?

Reply:
Created a function, `OpenApiPostUrlEncoded` to handle this on a lower level.